src/block.ml

```diff
@@ -147,10 +181,34 @@ module Pre = struct
         { blocks
         ; next = Rfenced_code (ind, num, q, info, Sub.to_string s :: lines, a)
         }
-    | Rdef_list (term, d :: defs), Lparagraph ->
+    | ( Rdef_list ({ empty_line_seen = false; _ } as deflist)
+      , Ldef_list (indent, def) ) ->
         { blocks
-        ; next = Rdef_list (term, (d ^ "\n" ^ Sub.to_string s) :: defs)
+        ; next =
+            Rdef_list
+              { deflist with
+                indent
+              ; defs = finish deflist.state :: deflist.defs
+              ; state = process empty (Sub.of_string def)
+              }
+        }
+    | Rdef_list deflist, Lempty ->
+        { blocks
+        ; next =
+            Rdef_list
+              { deflist with
+                empty_line_seen = true
+              ; state = process deflist.state s
+              }
         }
+    | Rdef_list deflist, _ when Parser.indent s >= deflist.indent ->
+        let s = Sub.offset deflist.indent s in
+        let state = process deflist.state s in
+        { blocks; next = Rdef_list { deflist with state } }
```

Comment:
I guess this should set `empty_line_seen = false` to fix the problem with lazy wrapping you mention below.

Reply:
This does address the lazy wrapping, thanks for the tip! Unfortunately it also breaks the nested def-list parsing. I'll continue considering it, but if an immediate solution occurs to you, let me know ;)

Reply:
Reordering the clauses such that the one that checks for the indentation level comes first solves the problem with the nested def-list test. The only remaining problem on def_list-003 test then are missing `<p>`s in the generated html.